# Lab | Hypothesis Testing

**Objective**

Welcome to the Hypothesis Testing Lab, where we embark on an enlightening journey through the realm of statistical decision-making! In this laboratory, we delve into various scenarios, applying the powerful tools of hypothesis testing to scrutinize and interpret data.

From testing the mean of a single sample (One Sample T-Test), to investigating differences between independent groups (Two Sample T-Test), and exploring relationships within dependent samples (Paired Sample T-Test), our exploration knows no bounds. Furthermore, we'll venture into the realm of Analysis of Variance (ANOVA), unraveling the complexities of comparing means across multiple groups.

So, grab your statistical tools, prepare your hypotheses, and let's embark on this fascinating journey of exploration and discovery in the world of hypothesis testing!

**Challenge 1**

In this challenge, we will be working with pokemon data. The data can be found here:

- https://raw.githubusercontent.com/data-bootcamp-v4/data/main/pokemon.csv

In [2]:
#libraries
import pandas as pd
import scipy.stats as st
import numpy as np



In [3]:
df = pd.read_csv("https://raw.githubusercontent.com/data-bootcamp-v4/data/main/pokemon.csv")
df

,Name,Type 1,Type 2,HP,Attack,Defense,Sp. Atk,Sp. Def,Speed,Generation,Legendary
0,Bulbasaur,Grass,Poison,45,49,49,65,65,45,1,False
1,Ivysaur,Grass,Poison,60,62,63,80,80,60,1,False
2,Venusaur,Grass,Poison,80,82,83,100,100,80,1,False
3,Mega Venusaur,Grass,Poison,80,100,123,122,120,80,1,False
4,Charmander,Fire,NaN,39,52,43,60,50,65,1,False
...,...,...,...,...,...,...,...,...,...,...,...
795,Diancie,Rock,Fairy,50,100,150,100,150,50,6,True
796,Mega Diancie,Rock,Fairy,50,160,110,160,110,110,6,True
797,Hoopa Confined,Psychic,Ghost,80,110,60,150,130,70,6,True
798,Hoopa Unbound,Psychic,Dark,80,160,60,170,130,80,6,True


- We posit that Pokemons of type Dragon have, on average, more HP stats than Grass. Choose the propper test and, with 5% significance, comment your findings.

In [5]:
#filtering the data including the pokemon with type "Dragon" and "Grass"
dragon_hp = df[df['Type 1'] == 'Dragon']['HP']
grass_hp = df[df['Type 1'] == 'Grass']['HP']
print(f"Dragon — count: {len(dragon_hp)}, media HP: {dragon_hp.mean():.2f}")
print(f"Grass — count: {len(grass_hp)}, media HP: {grass_hp.mean():.2f}")

Dragon — count: 32, media HP: 83.31
Grass — count: 70, media HP: 67.27


In [6]:
#test tt.ttest
t_statistic, p_value = st.ttest_ind(dragon_hp, grass_hp,alternative='greater')
print(f"T-statistic: {t_statistic:.4f}, P-value: {p_value:.4f}")

T-statistic: 3.5904, P-value: 0.0003


* H0: el HP medio de los Pokémon tipo Dragon es igual al HP medio de los tipo Grass.
* H1: el HP medio de los Pokémon tipo Dragon es mayor que el de los tipo Grass.
Has elegido correctamente un two-sample t-test porque estás comparando la media de una variable numérica (HP) entre dos grupos independientes.
Los datos descriptivos muestran:

* Dragon → n = 32, media HP = 83.31
* Grass → n = 70, media HP = 67.27
La diferencia observada es de aproximadamente:
83.31−67.27=16.0483.31 - 67.27 = 16.0483.31−67.27=16.04
Conclusión en términos de negocio/juego
Los resultados sugieren que los Pokémon de tipo Dragon tienen, en promedio, una mayor cantidad de HP que los Pokémon de tipo Grass. Esto implica que los Dragon suelen ser más resistentes en combate y pueden permanecer más tiempo en batalla antes de debilitarse.
Desde una perspectiva de diseño y balance del juego:

* Los Pokémon Dragon parecen estar diseñados como criaturas más poderosas y difíciles de derrotar.
* Los Pokémon Grass tienden a tener menos resistencia y probablemente dependen más de estrategias, estados alterados o ventajas de tipo en lugar de aguantar daño directo.
* Para un jugador competitivo, elegir Pokémon Dragon puede aportar ventaja en combates largos gracias a su mayor supervivencia.
* Para los desarrolladores, esta diferencia confirma una posible intención de balance: los tipos Dragon suelen ser considerados raros y fuertes, por lo que tienen estadísticas base más altas.

- We posit that Legendary Pokemons have different stats (HP, Attack, Defense, Sp.Atk, Sp.Def, Speed) when comparing with Non-Legendary. Choose the propper test and, with 5% significance, comment your findings.


**Challenge 2**

In this challenge, we will be working with california-housing data. The data can be found here:
- https://raw.githubusercontent.com/data-bootcamp-v4/data/main/california_housing.csv

In [11]:
df = pd.read_csv("https://raw.githubusercontent.com/data-bootcamp-v4/data/main/california_housing.csv")
df.head()

,longitude,latitude,housing_median_age,total_rooms,total_bedrooms,population,households,median_income,median_house_value
0,-114.31,34.19,15.0,5612.0,1283.0,1015.0,472.0,1.4936,66900.0
1,-114.47,34.40,19.0,7650.0,1901.0,1129.0,463.0,1.8200,80100.0
2,-114.56,33.69,17.0,720.0,174.0,333.0,117.0,1.6509,85700.0
3,-114.57,33.64,14.0,1501.0,337.0,515.0,226.0,3.1917,73400.0
4,-114.57,33.57,20.0,1454.0,326.0,624.0,262.0,1.9250,65500.0


In [12]:
print(df.columns.tolist())

['longitude', 'latitude', 'housing_median_age', 'total_rooms', 'total_bedrooms', 'population', 'households', 'median_income', 'median_house_value']


**We posit that houses close to either a school or a hospital are more expensive.**

- School coordinates (-118, 34)
- Hospital coordinates (-122, 37)

We consider a house (neighborhood) to be close to a school or hospital if the distance is lower than 0.50.

Hint:
- Write a function to calculate euclidean distance from each house (neighborhood) to the school and to the hospital.
- Divide your dataset into houses close and far from either a hospital or school.
- Choose the propper test and, with 5% significance, comment your findings.
 

In [7]:
# Distancia euclídea
def euclidean_distance(x1, y1, x2, y2):
    return np.sqrt((x2 - x1) ** 2 + (y2 - y1) ** 2)



In [14]:
# Distancia school and hospital
df['dist_school'] = df.apply(lambda row: euclidean_distance(row['longitude'], row['latitude'], -118, 34), axis=1)
df['dist_hospital'] = df.apply(lambda row: euclidean_distance(row['longitude'], row['latitude'], -122, 37), axis=1)
print(df[['dist_school', 'dist_hospital']].head())


   dist_school  dist_hospital
0     3.694888       8.187319
1     3.552591       7.966235
2     3.453940       8.143077
3     3.448840       8.154416
4     3.456848       8.183508


In [15]:
df['is_close'] = (df['dist_school'] < 0.50) | (df['dist_hospital'] < 0.50)
print(df['is_close'].value_counts())

is_close
False    10171
True      6829
Name: count, dtype: int64


In [17]:
houses_closer = df[df['is_close'] == True]['median_house_value']
houses_farther = df[df['is_close'] == False]['median_house_value']
print(f"Close — count: {len(houses_closer)}, media house value: {houses_closer.mean():.2f}")
print(f"Farther — count: {len(houses_farther)}, media house value: {houses_farther.mean():.2f}")

Close — count: 6829, media house value: 246951.98
Farther — count: 10171, media house value: 180678.44


In [18]:
#Two sample t-test
t_statistic, p_value = st.ttest_ind(houses_closer, houses_farther, alternative='greater')
print(f"T-statistic: {t_statistic:.4f}, P-value: {p_value:.4f}")


T-statistic: 38.0463, P-value: 0.0000


Los resultados del análisis estadístico muestran una diferencia clara entre el precio medio de las viviendas cercanas a una escuela o un hospital y aquellas situadas más lejos. Las casas ubicadas cerca de estos servicios presentan un precio medio de aproximadamente 246.951 dólares, mientras que las viviendas alejadas tienen un precio medio de 180.678 dólares. Esto supone una diferencia aproximada de 66.273 dólares a favor de las propiedades cercanas.
El contraste de hipótesis realizado mediante un Two Sample T-Test de una cola obtuvo un estadístico t de 38.0463 y un p-value prácticamente igual a 0. Debido a que este valor es muy inferior al nivel de significación habitual (0,05), se rechaza la hipótesis nula (H0) y se acepta la hipótesis alternativa (H1). Por tanto, existe evidencia estadísticamente significativa para afirmar que las viviendas cercanas a una escuela o un hospital son más caras que aquellas que se encuentran alejadas de estos servicios.
Desde una perspectiva de negocio, este resultado tiene implicaciones relevantes para el mercado inmobiliario. Para los compradores, confirma que la proximidad a servicios esenciales incrementa el valor de una propiedad, ya que estas ubicaciones ofrecen mayor comodidad, accesibilidad y calidad de vida. Para los vendedores, disponer de una vivienda cerca de escuelas u hospitales representa una ventaja competitiva que puede justificar precios más elevados y facilitar la venta. En el caso de los inversores inmobiliarios, el análisis sugiere que las zonas próximas a infraestructuras clave pueden ofrecer mayores oportunidades de revalorización y rentabilidad a largo plazo, convirtiéndose en áreas estratégicas para futuras inversiones.